yt/utilities/on_demand_imports.py

```diff
@@ -231,359 +136,11 @@ def __version__(self):
-            except ImportError:
-                h5f = NotAModule(self._name)
-            self._h5f = h5f
-        return self._h5f
-
-    _h5p = None
-
-    @property
-    def h5p(self):
-        if self._err:
-            raise self._err
-        if self._h5p is None:
-            try:
-                import h5py.h5p as h5p
-            except ImportError:
-                h5p = NotAModule(self._name)
-            self._h5p = h5p
-        return self._h5p
-
-    _h5d = None
-
-    @property
-    def h5d(self):
-        if self._err:
-            raise self._err
-        if self._h5d is None:
-            try:
-                import h5py.h5d as h5d
-            except ImportError:
-                h5d = NotAModule(self._name)
-            self._h5d = h5d
-        return self._h5d
-
-    _h5s = None
-
-    @property
-    def h5s(self):
-        if self._err:
-            raise self._err
-        if self._h5s is None:
-            try:
-                import h5py.h5s as h5s
-            except ImportError:
-                h5s = NotAModule(self._name)
-            self._h5s = h5s
-        return self._h5s
-
-    _version = None
-
-    @property
-    def version(self):
-        if self._err:
-            raise self._err
-        if self._version is None:
-            try:
-                import h5py.version as version
-            except ImportError:
-                version = NotAModule(self._name)
-            self._version = version
-        return self._version
-
-
-_h5py = h5py_imports()
-
-
-class nose_imports:
-    _name = "nose"
-    _run = None
-
-    @property
-    def run(self):
-        if self._run is None:
-            try:
-                from nose import run
-            except ImportError:
-                run = NotAModule(self._name)
-            self._run = run
-        return self._run
-
-
-_nose = nose_imports()
-
-
-class libconf_imports:
-    _name = "libconf"
-    _load = None
-
-    @property
-    def load(self):
-        if self._load is None:
-            try:
-                from libconf import load
-            except ImportError:
-                load = NotAModule(self._name)
-            self._load = load
-        return self._load
-
-
-_libconf = libconf_imports()
-
-
-class yaml_imports:
-    _name = "yaml"
-    _load = None
-    _FullLoader = None
-
-    @property
-    def load(self):
-        if self._load is None:
-            try:
-                from yaml import load
-            except ImportError:
-                load = NotAModule(self._name)
-            self._load = load
-        return self._load
-
-    @property
-    def FullLoader(self):
-        if self._FullLoader is None:
-            try:
-                from yaml import FullLoader
-            except ImportError:
-                FullLoader = NotAModule(self._name)
-            self._FullLoader = FullLoader
-        return self._FullLoader
-
-
-_yaml = yaml_imports()
-
-
 class NotMiniball(NotAModule):
```

Comment:
What is special about Miniball?

Reply:
The error message is unique. I left it for later but I'll get to it (and cartopy as well)